src/gui/screen_menu_lan_settings.cpp

```diff
@@ -23,7 +23,14 @@ ScreenMenuConnectionBase::ScreenMenuConnectionBase(uint32_t dev_id, const char *
 
 void ScreenMenuConnectionBase::refresh_addresses() {
     char str[ADDR_LEN];
-
+    char fwver[4];
+    const uint8_t fw_version = espif_firmware_version();
+    if (fw_version > 0) {
+        snprintf(fwver, sizeof(fwver), "%d", espif_firmware_version());
```

Comment:
You're calling the function here again even though you have the data stored in a variable

Reply:
Done.